### DISCUSSION TITLE
Using wildcards in StatAnalysis

### DISCUSSION THREAD

**Discussion by PerryShafran-NOAA** [dtcenter/METplus#1592] [Configuration] [answered]
Hi, there,

I am having issues running StatAnalysis using ENV and wildcards.  In my current example, I have the following variables set in the script before I call StatAnalysis.

export COMOUT=/lfs/h2/emc/vpppg/noscrub/Perry.Shafran/evs_output/v1.0/stats/cam
export model=namnest
export typ=grid2obs
export PDYm1=20220425
export regionnest=conusnest

In the StatAnalysis configuration file, I then have the following line:

MODEL1_STAT_ANALYSIS_LOOKIN_DIR = {ENV[COMOUT]}/ENV[model]}/{ENV[typ]}.{ENV[PDYm1]}/*{ENV[regionnest]}*

Now, this should translate into the following diretory:

/lfs/h2/emc/vpppg/noscrub/Perry.Shafran/evs_output/v1.0/stats/cam/namnest/grid2obs.20220425/*conusnest*

Which are actual files that exist.  However, I'm getting the following error in the log file:

04/26 13:52:04.110 metplus (stat_analysis_wrapper.py:1061) WARNING: Wildcard expansion found no matches
04/26 13:52:04.182 metplus (command_builder.py:255) ERROR: (stat_analysis_wrapper.py:1645) No value found for lookin dir
04/26 13:52:04Z jevs_namnest_grid2obs_stats_00-metplus: ERROR:  (stat_analysis_wrapper.py:1645) No value found for lookin dir

There is no printout of what directory/files that StatAnalysis thinks it is searching in to determine that there are no matches in the lookin dir.  That's not either in the log file or in the final configuration file.  Is there a way to determine what directory StatAnalysis thinks it is looking for?  Alternatively, can you help me figure out where the error exactly is here?

Thanks!

Perry

**Comment by jprestop** [comment, marked as answer]
Hi @PerryShafran-NOAA.  

Thank you for your question.  Setting LOG_LEVEL=DEBUG in the METplus configuration should show the substituted path that is used to find data.  Please give that a try and let us know how it goes.

Looking at what you listed here:

> MODEL1_STAT_ANALYSIS_LOOKIN_DIR = {ENV[COMOUT]}/ENV[model]}/{ENV[typ]}.{ENV[PDYm1]}/{ENV[regionnest]}

It looks like you may be missing an opening curly brace "{" before the ENV in "ENV[model]}".  I hope that helps!

Julie

**Reply by PerryShafran-NOAA** [reply]
Hi, @jprestop , Wow, I must have stared at that a thousand times and totally missed that curly bracket.  So hard to see the brackets and the curly braces when there is so many of them.  That has to be it.

I'll check that, thanks!

Perry

**Reply by jprestop** [reply]
Hi @PerryShafran-NOAA.  I just wanted to follow up and see if you were able to get this to work.  If so, could you please select "Mark as answer" and then I will lock this Discussion?  Thanks!

**Reply by PerryShafran-NOAA** [reply]
Yup, we are good,thanks!

Perry
